songhyun/app.js

```diff
@@ -42,6 +44,137 @@ app.post("/join", async (req, res) => {
     [name, password, email]
   );
   res.status(201).json({ message: "user created" });
+
+// 글 등록(과제3)
+app.post("/post", async (req, res) => {
+  const { title, content, user_id } = req.body;
+
+  await myDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);
+    `,
+    [title, content, user_id]
+  );
+  res.status(201).json({ message: "postCreated" });
+});
+
+// 글 보기(과제4)
+app.get("/post/view", async (req, res) => {
+  await myDataSource.query(
+    `SELECT
+      posts.user_id AS userId,
+      posts.id AS postingId,
+      posts.imageUrl AS postingImageUrl,
+      posts.content AS postingContent,
+      users.userProfileImage
+    FROM posts
+    INNER JOIN users ON posts.user_id = users.id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+// 유저가 작성한 글 보기(과제5)
+app.get("/user/view", async (req, res) => {
+  try {
+    const { user_id } = req.body;
+    const userInfo = await myDataSource.query(
+      `SELECT
+        users.id AS userId,
+        users.userProfileImage
+      FROM users
+      WHERE users.id=${user_id}`
+    );
+
+    const posting = await myDataSource.query(
+      `SELECT
+        posts.id AS postingId,
+        posts.imageUrl AS postingImageUrl,
+        posts.content AS postingContent
+      FROM posts, users
+      WHERE posts.user_id=users.id`
+    );
+    userInfo[0]["postings"] = posting;
+    res.status(200).json({ data: userInfo });
+  } catch (err) {
+    console.log("유저가 존재하지 않습니다.");
+  }
+});
+
+// 특정 글 내용 수정(과제6)
+app.patch("/post/update", async (req, res) => {
+  const { updatedContent } = req.body;
+
+  await myDataSource.query(
+    `UPDATE posts
+      SET content="${updatedContent}"
+      WHERE posts.id=1`
+  );
+
+  const showDB = await myDataSource.query(
+    `SELECT
+      users.id AS userId,
+      users.name AS userName,
+      posts.id AS postingId,
+      posts.title AS postingTitle,
+      posts.content AS postingContent
+    FROM users
+    INNER JOIN posts
+    ON users.id = posts.user_id`
+  );
+
+  res.status(201).json({ data: showDB });
+});
+
+// 특정 글 삭제(과제7)
+app.delete("/post/delete", async (req, res) => {
+  const { post_id } = req.body;
+
+  await myDataSource.query(
+    `DELETE FROM posts
+    WHERE posts.id = ${post_id}`,
+    () => {
+      res.status(200).json({ message: "postDeleted" });
+    }
+  );
+});
+
+// 좋아요 추가(과제8)
+app.post("/post/likes", async (req, res) => {
+  const { user_id, post_id } = req.body;
```

Comment:
```suggestion
  const { userId, postId } = req.body;
```
- JavaScript에서는 기본적으로 `snake_case`가 아닌 `CamelCase`로 작성합니다.
- 또한 snake_case, CamelCase를 뒤죽박죽 섞어서 사용하면 안됩니다.
- 하나의 컨벤션으로 통일해서 작성해 주세요.